### PR TITLE
Entity update

### PR DESCRIPTION
**Pull request by remy-rm**
Update Branch entity to match to bdd

### CODE REVIEW

File: src/main/java/com/templateproject/api/entity/Allergen.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @Column(name = "allergen_name")
+    private String allergenName;
+
+    // Getters and setters
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/entity/Allergen.java]: never put a setter for id or primary key

File: src/main/java/com/templateproject/api/entity/Category.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    private String categoryName;
+
+    // Getters and setters
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/entity/Category.java]: Remove setter for primary keys

File: src/main/java/com/templateproject/api/entity/Diet.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+@Entity
+public class Diet {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @Column(name = "diet_name")
+    private String dietName;
+
+    // Getters and setters
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/entity/Diet.java:16]: remove dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Comment by remy-rm** [comment]
clean code and add allergen, diet step and ingredient to recipe entity

**Review by ArPimentel — APPROVED** [review]

**Pull request closed without merging**